### PR TITLE
HOTFIX change spec.js style for fake client mode

### PR DESCRIPTION
**Pull request by pahaz**
Hote: you need to check SALES_SRM and FIREBASE config on production

### CODE REVIEW

**Review by AntonAL — APPROVED** [review]

**Review by sitozzz — APPROVED** [review]

File: apps/condo/domains/common/utils/GraphQLLoggerApp.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+const express = require('express')
+const gql = require('graphql-tag')
+const { get, set } = require('lodash')
+
+function normalizeQuery (string) {
+    if (!string) return ''
+    return string.replace(/[\s,]+/g, ' ').trim()
```
Comment (akarjakin) [inline apps/condo/domains/common/utils/GraphQLLoggerApp.js:7]: it's better for performance to move static regexps to constants and reuse.

File: apps/condo/domains/common/utils/GraphQLLoggerApp.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+function normalizeVariables (object) {
+    if (!object) return undefined
+    const data = JSON.parse(JSON.stringify(object))
+    for (const key of ['secret', 'password', 'data.password', 'data.secret']) {
```
Comment (akarjakin) [inline apps/condo/domains/common/utils/GraphQLLoggerApp.js:13]: this array will be recreated on every execution of the function, better to move to constant outside of function

File: apps/condo/domains/common/utils/codegeneration/generate.server.utils.js [hunk 3]
```diff
@@ -1,11 +1,55 @@
...
 const conf = require('@core/config')
+const { getById } = require('@core/keystone/schema')
 
 const IS_DEBUG = conf.NODE_ENV === 'development' || conf.NODE_ENV === 'test'
 
 const isNotUndefined = (x) => typeof x !== 'undefined'
 
+function _throwIfError (context, errors, data, errorMessage) {
+    if (errors) {
+        if (errors.some(e => e.originalError && e.originalError.data)) {
```
Comment (akarjakin) [inline apps/condo/domains/common/utils/codegeneration/generate.server.utils.js:12]:
```
if (IS_DEBUG) {
    const errorsToShow = errors.filter(error => error.originalError && error.originalError.data)
    
    if (!isEmpty(errorsToShow)) console.warn(map(errorsToShow, 'originalError.data'))

    console.error(errors)
}
```

File: apps/condo/domains/common/utils/codegeneration/generate.server.utils.js [hunk 4]
```diff
@@ -1,11 +1,55 @@
...
+        error.reqId = get(context, 'req.id')
+        throw error
+    }
+    if (!data || typeof data !== 'object') {
+        throw new Error('wrong query result')
+    }
+}
+
+async function execGqlAsUser (context, user, { query, variables, errorMessage = '[error] Internal Exec as user GQL Error', authedListKey = 'User', dataPath = 'obj' }) {
+    if (!context) throw new Error('wrong context argument')
```
Comment (akarjakin) [inline apps/condo/domains/common/utils/codegeneration/generate.server.utils.js:28]: 'wrong context argument' -> 'missing context argument'

here and on

File: apps/condo/domains/common/utils/codegeneration/generate.test.utils.js [hunk 5]
```diff
@@ -1,8 +1,31 @@
...
+    if (get(context, 'variables')) {
+        msg += `VARIABLES: ${JSON.stringify(context.variables)}`
+    }
+    return msg
+}
+
+function throwIfError (data, errors, context = {}) {
     if (errors) {
-        const err = new Error('TestRequestError')
+        if (IS_DEBUG && errors.some(e => e.originalError.data || e.originalError.internalData)) {
```
Comment (akarjakin) [inline apps/condo/domains/common/utils/codegeneration/generate.test.utils.js:24]:
```
if (IS_DEBUG) {
    const errorsToShow = errors.filter(error => error.originalError && error.originalError.data)
    
    if (!isEmpty(errorsToShow)) console.warn(map(errorsToShow, 'originalError.data'))

    console.error(errors)
}
```

File: apps/condo/domains/organization/integrations/sbbol/sync/syncOrganization.spec.js [hunk 6]
```diff
@@ -3,26 +3,21 @@
...
+const { setFakeClientMode } = require('@core/keystone/test.utils')
 const { makeClientWithRegisteredOrganization } = require('@condo/domains/organization/utils/testSchema/Organization')
 const { makeClientWithNewRegisteredAndLoggedInUser } = require('@condo/domains/user/utils/testSchema')
-
-const { MockSbbolResponses } = require('./MockSbbolResponses')
 const { OrganizationEmployee: OrganizationEmployeeApi, Organization: OrganizationApi } = require('@condo/domains/organization/utils/serverSchema')
-const { getItem, updateItem } = require('@keystonejs/server-side-graphql-client')
 
-let keystone
+const { keystone } = require('../../../../../index')
```
Comment (akarjakin) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncOrganization.spec.js:13]: `const { keystone } = require('@app/condo/index')`

File: apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.spec.js [hunk 7]
```diff
@@ -2,15 +2,17 @@
...
 const { ServiceSubscription } = require('@condo/domains/subscription/utils/serverSchema')
-const { prepareKeystoneExpressApp, setFakeClientMode } = require('@core/keystone/test.utils')
+const { setFakeClientMode } = require('@core/keystone/test.utils')
 const { makeClientWithNewRegisteredAndLoggedInUser } = require('@condo/domains/user/utils/testSchema')
 const { registerNewOrganization } = require('@condo/domains/organization/utils/testSchema/Organization')
 const { find } = require('lodash')
 const { SUBSCRIPTION_TYPE, SUBSCRIPTION_TRIAL_PERIOD_DAYS } = require('@condo/domains/subscription/constants')
 const dayjs = require('dayjs')
 
+const { keystone } = require('../../../../../index')
```
Comment (akarjakin) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncSubscriptions.spec.js:13]: `const { keystone } = require('@app/condo/index')`

File: apps/condo/domains/organization/integrations/sbbol/sync/syncUser.spec.js [hunk 8]
```diff
@@ -2,25 +2,20 @@
...
+const { setFakeClientMode } = require('@core/keystone/test.utils')
 const { makeClientWithRegisteredOrganization } = require('@condo/domains/organization/utils/testSchema/Organization')
 const { MockSbbolResponses } = require('./MockSbbolResponses')
 const { User: UserApi } = require('@condo/domains/user/utils/serverSchema')
 const { OnBoarding: OnBoardingApi } = require('@condo/domains/onboarding/utils/serverSchema')
 const { getItem, getItems } = require('@keystonejs/server-side-graphql-client')
 const { makeClientWithResidentUser } = require('@condo/domains/user/utils/testSchema')
 
-let keystone
+const { keystone } = require('../../../../../index')
```
Comment (akarjakin) [inline apps/condo/domains/organization/integrations/sbbol/sync/syncUser.spec.js:13]: `@app/condo/index`

File: apps/condo/domains/organization/utils/serverSchema/Organization.js [hunk 9]
```diff
@@ -10,9 +10,10 @@ const { getById } = require('@core/keystone/schema')
 const { DEFAULT_ROLES } = require('@condo/domains/organization/constants/common')
 const { SBBOL_FINGERPRINT_NAME } = require('@condo/domains/organization/integrations/sbbol/common')
 
-let SALES_CRM_WEBHOOKS_URL = attempt(() => JSON.parse(config.SALES_CRM_WEBHOOKS_URL))
-if (isError(SALES_CRM_WEBHOOKS_URL) || !SALES_CRM_WEBHOOKS_URL.subscriptions || !SALES_CRM_WEBHOOKS_URL.organizations) {
-    SALES_CRM_WEBHOOKS_URL = null
+const logger = pino({ name: 'sales_crm', enabled: falsey(process.env.DISABLE_LOGGING) })
+const SALES_CRM_WEBHOOKS_URL = (config.SALES_CRM_WEBHOOKS_URL) ? JSON.parse(config.SALES_CRM_WEBHOOKS_URL) : null
+if (SALES_CRM_WEBHOOKS_URL && !SALES_CRM_WEBHOOKS_URL.subscriptions && !SALES_CRM_WEBHOOKS_URL.organizations) {
+    throw new Error('Wrong SALES_CRM_WEBHOOKS_URL value')
```
Comment (akarjakin) [inline apps/condo/domains/organization/utils/serverSchema/Organization.js:16]: won't this fail on build faze ?

File: apps/condo/domains/ticket/schema/TicketStatus.test.js [hunk 10]
```diff
@@ -214,29 +214,24 @@ describe('TicketStatus', () => {
...
-            }))
-            
-        const isTranslationCompleted = Object.values(statuses).every(status => translations.includes(status.name))
+        const statuses = await TicketStatus.getAll(admin, {
+            organization_is_null: true,
+            name_starts_with: 'ticket.status.',
+        })
 
-        expect(isTranslationCompleted).toBeTruthy()
+        expect(statusTranslations.length).toBeGreaterThanOrEqual(6)
```
Comment (akarjakin) [inline apps/condo/domains/ticket/schema/TicketStatus.test.js:231]: what is 6? where it comes from? maybe it's xxxx.length? please add some comments or name it

File: apps/condo/domains/user/schema/RegisterNewUserService.test.js [hunk 11]
```diff
@@ -50,6 +50,7 @@ describe('RegisterNewUserService', () => {
         const client = await makeClient()
         const phone = createTestLandlineNumber()
 
+        // TODO(pahaz): refactor this! use expect error function!
```
Comment (akarjakin) [inline apps/condo/domains/user/schema/RegisterNewUserService.test.js:53]: we use to add tickets in Jira for TODOs :)

File: apps/condo/domains/user/schema/User.test.js [hunk 12]
```diff
@@ -31,8 +31,8 @@ describe('SIGNIN', () => {
             'identity': DEFAULT_TEST_USER_IDENTITY,
             'secret': DEFAULT_TEST_USER_SECRET,
         })
-        expect(data.auth.user.id).toMatch(/[a-zA-Z0-9-_]+/)
         expect(errors).toEqual(undefined)
+        expect(data.obj.item.id).toMatch(/[a-zA-Z0-9-_]+/)
```
Comment (akarjakin) [inline apps/condo/domains/user/schema/User.test.js:35]: better to move static regexps to constants outside the function

File: apps/condo/domains/user/utils/testSchema/index.js [hunk 13]
```diff
@@ -20,15 +21,16 @@ const { RESET_USER_MUTATION } = require('@condo/domains/user/gql')
 const User = generateGQLTestUtils(UserGQL)
 const UserAdmin = generateGQLTestUtils(UserAdminGQL)
 
+const usAreaCodes = countryPhoneData[0].mobile_begin_with.filter(x => x.length === 3)
 const createTestEmail = () => ('test.' + getRandomString() + '@example.com').toLowerCase()
-const createTestPhone = () => faker.phone.phoneNumber('+79#########')
+const createTestPhone = () => faker.random.arrayElement([faker.phone.phoneNumber('+79#########'), `+1${faker.random.arrayElement(usAreaCodes)}${faker.phone.phoneNumber('#######')}`])
```
Comment (akarjakin) [inline apps/condo/domains/user/utils/testSchema/index.js:26]: looks like generator for random US phone number needs to be moved to separate named function in order to increase code readability

File: packages/@core.keystone/test.utils.js [hunk 14]
```diff
@@ -410,6 +414,30 @@ const getSchemaObject = async (schemaList, fields, where) => {
     return result.data.obj
 }
 
+async function waitFor (callback, options = { timeout: 2000, interval: 50 }) {
+    const timeout = options.timeout || 5000
+    const interval = options.interval || 50
+    let savedError = null
+
+    return new Promise((res, rej) => {
```
Comment (akarjakin) [inline packages/@core.keystone/test.utils.js:422]: `res` is confusing with result. please user `resolve` & `reject`

**Review by akarjakin — APPROVED** [review]

File: apps/condo/domains/user/schema/User.test.js (hunk 12 above)
Comment (pahaz) [reply apps/condo/domains/user/schema/User.test.js:35]: it's useless for reuse. It's ok to use such things inside the tests

File: packages/@core.keystone/test.utils.js (hunk 14 above)
Comment (pahaz) [reply packages/@core.keystone/test.utils.js:422]: it's a common style for promises

File: apps/condo/domains/common/utils/GraphQLLoggerApp.js (hunk 1 above)
Comment (pahaz) [reply apps/condo/domains/common/utils/GraphQLLoggerApp.js:7]: you don't need to reuse it. it's better to store a full function context inside if it's possible or near the function it's increase the readability

File: apps/condo/domains/organization/utils/serverSchema/Organization.js (hunk 9 above)
Comment (pahaz) [reply apps/condo/domains/organization/utils/serverSchema/Organization.js:16]: No failt at build time becuse there is no `SALES_CRM_WEBHOOKS_URL` at that step

**Review by dkoviazin — APPROVED** [review]

**Comment by akarjakin** [comment]
lgtm

**Review by AntonAL — COMMENTED** [review]
LGTM

**Pull request merged**